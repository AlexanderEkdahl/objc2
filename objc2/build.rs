fn main() {
    // The script doesn't depend on our code
    println!("cargo:rerun-if-changed=build.rs");

    #[cfg(feature = "exception")]
    {
        use std::env;

        if std::env::var("DOCS_RS").is_ok() {
            // docs.rs doesn't have clang, so skip building this. The
            // documentation will still work since it doesn't need to link.
            return;
        }
        println!("cargo:rerun-if-changed=extern/exception.m");

        let mut builder = cc::Build::new();
        builder.file("extern/exception.m");

        for flag in env::var("DEP_OBJC_0_2_CC_ARGS").unwrap().split(' ') {
            builder.flag(flag);
        }

        builder.compile("librust_objc_try_catch_exception.a");
    }
}
