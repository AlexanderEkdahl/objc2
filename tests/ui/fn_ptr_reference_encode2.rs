//! Extra test for `fn_ptr_reference_encode`
//! (They fail at different compilation passes).
use objc2::Encode;

fn main() {
    let encoding = <extern "C" fn(&i32) as Encode>::ENCODING;
}
