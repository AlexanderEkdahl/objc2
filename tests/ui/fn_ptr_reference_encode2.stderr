error: implementation of `Encode` is not general enough
 --> ui/fn_ptr_reference_encode2.rs:6:20
  |
6 |     let encoding = <extern "C" fn(&i32) as Encode>::ENCODING;
  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ implementation of `Encode` is not general enough
  |
  = note: `Encode` would have to be implemented for the type `for<'r> extern "C" fn(&'r i32)`
  = note: ...but `Encode` is actually implemented for the type `extern "C" fn(&'0 i32)`, for some specific lifetime `'0`
